app/models/censuses/census_report.rb

```diff
@@ -0,0 +1,286 @@
+
+        yesterday = item.second
+      end
+
+      datasets = [
+        { label: labels[0], data: first_dataset },
+        { label: labels[1], data: second_dataset },
+      ]
+
+      add_dimension(data_source_or_project_type, organization_id, project_id, datasets, dimension_label) unless first_dataset.empty?
+    end
+
+    private def add_dimension(data_source_or_project_type, organization_id, project_id, datasets, title)
+      @shape[data_source_or_project_type] ||= {}
+      @shape[data_source_or_project_type][organization_id] ||= {}
+      @shape[data_source_or_project_type][organization_id][project_id] ||= {}
+      @shape[data_source_or_project_type][organization_id][project_id][:datasets] = datasets
+      @shape[data_source_or_project_type][organization_id][project_id][:title] = {}
+      @shape[data_source_or_project_type][organization_id][project_id][:title][:display] = true
+      @shape[data_source_or_project_type][organization_id][project_id][:title][:text] = title
+      @shape
+    end
+
+    # Detail view
+
+    def detail_name(project_count, project_type, data_source_id, organization_id, project_id)
+      if project_type != 'all'
+        ptype = GrdaWarehouse::Hud::Project::PROJECT_GROUP_TITLES[project_type]
+        return "#{project_count_str(project_count, prefix: ptype)} on"
+      end
+
+      projects = project_count_str(project_count)
+      return "#{projects} from All Sources on" if data_source_id == 'all'
+
+      if organization_id == 'all'
+        data_source_name = GrdaWarehouse::DataSource.find(data_source_id.to_i).name
+        return "#{projects} from #{data_source_name} on"
+      end
+
+      organization_name = GrdaWarehouse::Hud::Organization.find(organization_id.to_i).name(@filter.user)
+      return "#{projects} from #{organization_name} on" if project_id == 'all'
+
+      project_name = GrdaWarehouse::Hud::Project.find(project_id.to_i).name(@filter.user)
+      "#{project_name} at #{organization_name} on"
+    end
+
+    def clients_for_date(date, project_type, data_source, organization, project, population)
+      known_sub_populations = GrdaWarehouse::ServiceHistoryEnrollment.known_standard_cohorts
+
+      raise "Population #{population} not defined" unless known_sub_populations.include?(population.to_sym)
+
+      columns = {
+        'LastName' => c_t[:LastName].to_sql,
+        'FirstName' => c_t[:FirstName].to_sql,
+        'ProjectName' => she_t[:project_name].to_sql,
+        'short_name' => ds_t[:short_name].to_sql,
+        'client_id' => she_t[:client_id].to_sql,
+        'project_id' => p_t[:id].to_sql,
+        'confidential' => bool_or(p_t[:confidential], o_t[:confidential]),
+      }
+
+      enrollments = GrdaWarehouse::ServiceHistoryEnrollment.residential.
+        send(population).
+        service_on_date(date).
+        joins(:client, :data_source, :organization, :project).
+        merge(census_projects_scope) # merge with project scope filtered by @filter params
```

Comment:
@eanders  need extra attention here. The counts are adding up correctly, so I think it's right, but I'm not very confident.

This basically needs to match whats in `GrdaWarehouse::Census::ProjectBatch.get_client_and_project_counts` so that the counts on the charts match up with the total count on the detail page.

Also - We already limit the `census_projects_scope` to residential, so that `residential` scope on SHE is redundant. Is limiting to residential projects for the census correct?